tests fail to link

I think a recent commit broke something that causes linker errors when running the tests for me. `git bisect` points to https://github.com/memorysafety/rav1d/commit/3bf213cde9c03fa8f13782445d1a1081cde8b176 as the cause. I'm on an x86_64 linux machine, and running the tests works fine before this commit.

```
folkertdev@folkertdev ~/r/r/build ((3bf213cd…)|BISECTING) [125]> cargo build --release && meson test \
                                                                   --suite testdata-8 \
                                                                   --suite testdata-10 \
                                                                   --suite testdata-12 \
                                                                   --suite testdata-multi
ninja: Entering directory `/home/folkertdev/rust/rav1d/build'
[2/3] Linking target tools/dav1d.
FAILED: tools/dav1d 
cc  -o tools/dav1d 'tools/f9d35d4@@dav1d@exe/dav1d.c.o' 'tools/f9d35d4@@dav1d@exe/dav1d_cli_parse.c.o' -Wl,--as-needed -Wl,--no-undefined -Wl,-O1 -Wl,--start-group librav1d.a tools/libdav1d_input.a tools/libdav1d_output.a -pthread -lm -Wl,--end-group '-Wl,-rpath,$ORIGIN/..:$ORIGIN/' -Wl,-rpath-link,/home/folkertdev/rust/rav1d/build/ -Wl,-rpath-link,/home/folkertdev/rust/rav1d/build/tools
/usr/bin/ld: librav1d.a(std-eb8a07e9874efeeb.std.ee38bba0a237925c-cgu.0.rcgu.o): in function `std::sys::unix::weak::fetch':
/rustc/6b771f6b5a6c8b03b6322a9c77ac77cb346148f0/library/std/src/sys/unix/weak.rs:138: undefined reference to `dlsym'
collect2: error: ld returned 1 exit status
[3/3] Linking target tests/seek_stress.
FAILED: tests/seek_stress 
cc  -o tests/seek_stress 'tests/59830eb@@seek_stress@exe/seek_stress.c.o' 'tools/f9d35d4@@dav1d@exe/dav1d_cli_parse.c.o' 'tools/f9d35d4@@dav1d_input@sta/input_input.c.o' 'tools/f9d35d4@@dav1d_input@sta/input_ivf.c.o' -Wl,--as-needed -Wl,--no-undefined -Wl,-O1 -Wl,--start-group librav1d.a -pthread -lm -Wl,--end-group '-Wl,-rpath,$ORIGIN/..' -Wl,-rpath-link,/home/folkertdev/rust/rav1d/build/
/usr/bin/ld: librav1d.a(std-eb8a07e9874efeeb.std.ee38bba0a237925c-cgu.0.rcgu.o): in function `std::sys::unix::weak::fetch':
/rustc/6b771f6b5a6c8b03b6322a9c77ac77cb346148f0/library/std/src/sys/unix/weak.rs:138: undefined reference to `dlsym'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
Could not rebuild /home/folkertdev/rust/rav1d/build
```

tagging @thedataking

--- a/README.md
+++ b/README.md
@@ -8,26 +8,23 @@ rav1d is currently experimental. Core functionality has been transpiled using [c
 
 Currently we use the original Meson test suite for testing the Rust port. To setup and run these tests, do the following:
 
-First, build the Rust project using Cargo. You'll need to do this step manually before running any tests because it is not built automatically when tests are run. Note that you need to build with the `--release` flag, as tests are run against the release binary.
+First, build the Rust project using Cargo. You'll need to do this step manually before running any tests because it is not built automatically when tests are run. Note that we build with the `--release` flag, adjust paths accordingly
+for debug or cross-target builds.
 
 ```txt
 cargo build --release
 ```
 
-Then create the `build` dir and run `meson setup` in it:
+Second, create the `build` dir with `meson`:
 
 ```txt
-mkdir build
-cd build
-meson setup ..
+meson setup build
 ```
 
-Then you can run `meson test` to run the tests. Currently only the `testdata-*` suites are setup to test against the Rust executable:
+Then you can run the tests with:
 
 ```txt
-meson test \
-  --suite testdata-8 \
-  --suite testdata-10 \
-  --suite testdata-12 \
-  --suite testdata-multi
+.github/workflows/test.sh -r target/release/dav1d
 ```
+
+You can learn more about how to build and test in the `.github/workflows` folder.

--- a/tests/meson.build
+++ b/tests/meson.build
@@ -146,15 +146,16 @@ if get_option('enable_tools')
 endif
 
 if get_option('test_rust')
+    # options to test with out-of-tree versions of dav1d and seek_stress
     test_rust_path = get_option('test_rust_path')
     seek_stress_test_rust_path = get_option('seek_stress_test_rust_path')
 
-    # We're trying to test the Rust build, override the `dav1d` and
-    # `seek_stress` executables to point to the versions given during setup.
+    # Adjust relative paths so one can use binary paths relative to project
+    # top dir rather than relative to the location of the test subdirectory.
     if test_rust_path != ''
-        # Adjust relative paths so one can use binary paths relative to project
-        # top dir rather than relative to the location of the test subdirectory.
         dav1d = files(join_paths('../', get_option('test_rust_path')))
+    endif
+    if seek_stress_test_rust_path != ''
         seek_stress = files(join_paths('../', get_option('seek_stress_test_rust_path')))
     endif
 endif

